cylc/flow/scheduler.py

```diff
@@ -621,7 +570,10 @@ async def run_scheduler(self):
                     self
                 )
             )
-            await self.publisher.publish(self.data_store_mgr.publish_deltas)
+            self.server.publish_queue.put(
+                self.data_store_mgr.publish_deltas)
+            # Yield control to other threads
+            sleep(0)
```

Comment:
As above
```suggestion
            await asyncio.sleep(0)
```

Reply:
`time.sleep` is used to yield control to other threads (halting the current thread).. Whereas, I believe `asyncio.sleep` just allows other ayncio execution for the duration.